core/src/main/java/org/elasticsearch/search/suggest/SuggestUtils.java

```diff
@@ -208,78 +202,6 @@ public static StringDistance resolveDistance(String distanceVal) {
-                    throw new IllegalArgumentException("Illegal max_edits value " + suggestion.maxEdits());
-                }
-            } else if (parseFieldMatcher.match(fieldName, Fields.MAX_INSPECTIONS)) {
-                suggestion.maxInspections(parser.intValue());
-            } else if (parseFieldMatcher.match(fieldName, Fields.MAX_TERM_FREQ)) {
-                suggestion.maxTermFreq(parser.floatValue());
-            } else if (parseFieldMatcher.match(fieldName, Fields.PREFIX_LENGTH)) {
-                suggestion.prefixLength(parser.intValue());
-            } else if (parseFieldMatcher.match(fieldName, Fields.MIN_WORD_LENGTH)) {
-                suggestion.minWordLength(parser.intValue());
-            } else if (parseFieldMatcher.match(fieldName, Fields.MIN_DOC_FREQ)) {
-                suggestion.minDocFreq(parser.floatValue());
-            } else {
-                return false;
-            }
-            return true;
-    }
-
-    public static boolean parseSuggestContext(XContentParser parser, MapperService mapperService, String fieldName,
-            SuggestionSearchContext.SuggestionContext suggestion, ParseFieldMatcher parseFieldMatcher) throws IOException {
-
-        if (parseFieldMatcher.match(fieldName, Fields.ANALYZER)) {
-            String analyzerName = parser.text();
-            Analyzer analyzer = mapperService.analysisService().analyzer(analyzerName);
-            if (analyzer == null) {
-                throw new IllegalArgumentException("Analyzer [" + analyzerName + "] doesn't exists");
-            }
-            suggestion.setAnalyzer(analyzer);
-        } else if (parseFieldMatcher.match(fieldName, Fields.FIELD)) {
-            suggestion.setField(parser.text());
-        } else if (parseFieldMatcher.match(fieldName, Fields.SIZE)) {
-            suggestion.setSize(parser.intValue());
-        } else if (parseFieldMatcher.match(fieldName, Fields.SHARD_SIZE)) {
-            suggestion.setShardSize(parser.intValue());
-        } else {
-           return false;
-        }
-        return true;
-    }
-
-    public static void verifySuggestion(MapperService mapperService, BytesRef globalText, SuggestionContext suggestion) {
-        // Verify options and set defaults
-        if (suggestion.getField() == null) {
-            throw new IllegalArgumentException("The required field option is missing");
-        }
-        if (suggestion.getText() == null) {
-            if (globalText == null) {
-                throw new IllegalArgumentException("The required text option is missing");
-            }
-            suggestion.setText(globalText);
-        }
-        if (suggestion.getAnalyzer() == null) {
-            suggestion.setAnalyzer(mapperService.searchAnalyzer());
-        }
-        if (suggestion.getShardSize() == -1) {
-            suggestion.setShardSize(Math.max(suggestion.getSize(), 5));
-        }
```

Comment:
Those two checks are part of `SuggestionBuilder#populateCommonFields()` and are called from the `build()` method on the shard.